src/tpt_reports/report_generator.py

```diff
@@ -0,0 +1,423 @@
+"""Generate a TPT report using a passed data dictionary."""
+
+# Standard Python Libraries
+from datetime import datetime
+import os
+import secrets
+
+# Third-Party Libraries
+import numpy as np
+import pandas as pd
+from reportlab.lib import utils
+from reportlab.lib.colors import HexColor
+from reportlab.lib.pagesizes import letter
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.lib.units import inch
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from reportlab.platypus import (
+    Image,
+    KeepTogether,
+    PageBreak,
+    Paragraph,
+    Spacer,
+    Table,
+    TableStyle,
+)
+from reportlab.platypus.doctemplate import (
+    BaseDocTemplate,
+    NextPageTemplate,
+    PageTemplate,
+)
+from reportlab.platypus.frames import Frame
+
+BASE_DIR = os.path.abspath(os.path.dirname(__file__))
+TODAYS_DATE = datetime.today().strftime("%m/%d/%Y")
+
+# Set fonts to be used in the pdf
+pdfmetrics.registerFont(
+    TTFont("Franklin_Gothic_Book", f"{BASE_DIR}/fonts/FranklinGothicBook.ttf")
+)
+
+pdfmetrics.registerFont(
+    TTFont(
+        "Franklin_Gothic_Medium_Regular",
+        f"{BASE_DIR}/fonts/FranklinGothicMediumRegular.ttf",
+    )
+)
+
+# Set your page size
+defaultPageSize = letter
+PAGE_HEIGHT = defaultPageSize[1]
+PAGE_WIDTH = defaultPageSize[0]
+
+
+# Issue #23 - test generate_password()
+# TODO: Add unit tests for following logic and remove this comment.
+def generate_password(length):
+    """Generate a string for the password for the pdf."""
+    # Generate a random string of hexadecimal digits
+    password = secrets.token_hex(length // 2)
+
+    # Return the first `length` characters of the password
+    return password[:length]
+
+
+class MyDocTemplate(BaseDocTemplate):
+    """Extend the BaseDocTemplate to adjust Template."""
+
+    def __init__(self, filename, encrypt, **kw):
+        """Initialize MyDocTemplate."""
+        # Set Key as ENV variable
+        os.environ["TPT_REPORT_KEY"] = encrypt
+        self.allowSplitting = 0
+        BaseDocTemplate.__init__(self, filename, **kw)
+        self.pagesize = defaultPageSize
+        self.encrypt = encrypt
```

Comment:
The `encrypt` variable is confusingly named.  A variable named `encrypt` would typically be a boolean for whether or not to encrypt the reports.  How about `report_key` instead?
```suggestion
    def __init__(self, filename, report_key, **kw):
        """Initialize MyDocTemplate."""
        # Set report_key as an environment variable
        os.environ["TPT_REPORT_KEY"] = report_key
        self.allowSplitting = 0
        BaseDocTemplate.__init__(self, filename, **kw)
        self.pagesize = defaultPageSize
        self.report_key = report_key
```